openmdao/approximation_schemes/complex_step.py

```diff
@@ -173,96 +112,42 @@ def compute_approximations(self, system, jac, total=False):
 
-        if total:
-            current_vec = system._outputs
-        else:
-            current_vec = system._residuals
-
-        # Clean vector for results
-        results_clone = current_vec._clone(True)
-
         # Turn on complex step.
         system._set_complex_step_mode(True)
-        results_clone.set_complex_step_mode(True)
-
-        # To support driver src_indices, we need to override some checks in Jacobian, but do it
-        # selectively.
-        uses_src_indices = (system._owns_approx_of_idx or system._owns_approx_wrt_idx) and \
-            not isinstance(jac, dict)
-
-        use_parallel_fd = system._num_par_fd > 1 and (system._full_comm is not None and
-                                                      system._full_comm.size > 1)
-        num_par_fd = system._num_par_fd if use_parallel_fd else 1
-        is_parallel = use_parallel_fd or system.comm.size > 1
-
-        results = defaultdict(list)
-        iproc = system.comm.rank
-        owns = system._owning_rank
-        mycomm = system._full_comm if use_parallel_fd else system.comm
-
-        fd_count = 0
-        approx_groups = self._get_approx_groups(system)
-        for tup in approx_groups:
-            wrt, delta, fact, in_idx, in_size, outputs = tup
-            for i_count, idx in enumerate(in_idx):
-                if fd_count % num_par_fd == system._par_fd_id:
-                    # Run the Finite Difference
-                    result = self._run_point_complex(system, wrt, idx, delta, results_clone, total)
-
-                    if is_parallel:
-                        for of, _, out_idx in outputs:
-                            if owns[of] == iproc:
-                                results[(of, wrt)].append(
-                                    (i_count, result._views_flat[of][out_idx].imag.copy()))
-                    else:
-                        for of, subjac, out_idx in outputs:
-                            subjac[:, i_count] = result._views_flat[of][out_idx].imag
-
-                fd_count += 1
-
-        if is_parallel:
-            results = _gather_jac_results(mycomm, results)
-
-        for wrt, _, fact, _, _, outputs in approx_groups:
-            for of, subjac, _ in outputs:
-                key = (of, wrt)
-                if is_parallel:
-                    for i, result in results[key]:
-                        subjac[:, i] = result
-
-                subjac *= fact
-                if uses_src_indices:
-                    jac._override_checks = True
-                    jac[key] = subjac
-                    jac._override_checks = False
-                else:
-                    jac[key] = subjac
+
+        self._compute_approximations(system, jac, total, under_cs=True)
 
         # Turn off complex step.
         system._set_complex_step_mode(False)
 
-    def _run_point_complex(self, system, in_name, idxs, delta, result_clone, total=False):
+    def _get_multiplier(self, delta):
+        return (1.0 / delta * 1j).real
+
```

Comment:
These two functions could use doc strings to make things more clear. Especially get_multiplier.

Reply:
done